### PR TITLE
smallest positive integer program

### PR DESCRIPTION
**Pull request by joynewton1996**

### CODE REVIEW

File: hacker_Rank/May8_Answer.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        return [1]
+
+
+def sequence_generator(n):
+    sequence = []
+    step = 1
+    while step <= n:
+        sequence.append(step)
+        step += 1
+    return sequence
```
Comment (roy-einstein) [inline hacker_Rank/May8_Answer.py]: Why do you need this ?

File: hacker_Rank/May8_Answer.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+obj = Algo()
+
+
+def solution(a):
+    if a >= [0]:
+        Max_value = obj.max_value(a)
+        sequence = sequence_generator(Max_value)
+        result = []
+        for value in sequence:
```
Comment (roy-einstein) [inline hacker_Rank/May8_Answer.py]: sequence can be = range(1,Max_value+1) ???

File: hacker_Rank/May8_Answer.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                result.append(value)
+
+        if result > []:
+            result = obj.sort(result)
+            return result
+        else:
+            result.append(Max_value + 1)
+            return result
+    else:
+        return [1]
```
Comment (roy-einstein) [inline hacker_Rank/May8_Answer.py]: A= [1,3,6,4,5]
ouput == 2,
loop thru A, if you find smallest missing element break the loop and send ``` 1 ```
A=[1,2,3,4]
output is -1,
if series and given A is same, then return -1

**Review by roy-einstein — CHANGES_REQUESTED** [review]
Try to USE ``` Sets ``` , intersection, difference, which can be solved very easily with these

File: hacker_Rank/May8_Answer.py (hunk 2 above)
Comment (joynewton1996) [reply hacker_Rank/May8_Answer.py]: yes. got the point

File: hacker_Rank/May8_Answer.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from algorithm import Algo
+
+obj = Algo()
+
+
+def solution(a):
+    if a >= [0]:
+        Max_value = obj.max_value(a)
```
Comment (roy-einstein) [inline hacker_Rank/May8_Answer.py]: Why variable name ``` Max_value ``` capital in it

File: hacker_Rank/May8_Answer.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        sequence = {item for item in range(1, Max_value + 2)}
+        a = set(a)
+        print(a)
+        difference = sequence.difference(a)
+        difference = list(difference)
+        return difference[0]
```
Comment (roy-einstein) [inline hacker_Rank/May8_Answer.py]: ``` return difference[0] ``` how do you [0] element is always a smaller

File: hacker_Rank/May8_Answer.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+obj = Algo()
+
+
+def solution(a):
+    if a >= [0]:
+        Max_value = obj.max_value(a)
+        sequence = {item for item in range(1, Max_value + 2)}
+        a = set(a)
+        print(a)
```
Comment (roy-einstein) [inline hacker_Rank/May8_Answer.py]: why do need to convert into set...?

**Review by roy-einstein — CHANGES_REQUESTED** [review]
need some more work

**Review by roy-einstein — CHANGES_REQUESTED** [review]
needs work

**Review by roy-einstein — APPROVED** [review]
Keep good work

**Pull request merged**